### ISSUE TITLE
Consider removing "allow_duplicate_filenames" config option

### ISSUE THREAD

**Issue by cmkohnen** [SjoerdB93/Graphs#26]
As the title says.
Whilst discovering and trying to fix a logic error in add_equation.py, I discovered that although multiple files can have the same name, only the latest added would be considered in the graph or the datadict, overriding any previous one. 

The related code is in add_equation.py at line 23 and datman.py at line 152

The function of just adding a number in brackets works just fine, maybe adding a toast after addition saying the name was changed would be optimal.

Correct me if I'm wrong

**Comment by SjoerdB93** [comment]
I've just checked this, and it seems something has been broken along the way. 

Originally, this started as a tool that I just used myself to make my life easier, and then grew into a bit more proper hobby project. 
In the very original design, it only loaded new data if the filename didn't exist. So if I tried to open a datafile that was already open, it didn't open it again. This was the default option.

Along the way, I added the option to open the same data multiple times anyway, and just made it add a number in brackets afterwards. As this is probably the most sensible default for most people, I made this the default setting. So by default, "allow_duplicate_filenames" is true. It's also how I have it configured myself usually, so I'm not sure when the "false" functionality broke. 

So the idea is that if you set this config option to false, and then import a dataset twice, that it doesn't load this dataset again. But this feature mainly grew out of the fact that in my initial prototype, I ran into issues with duplicate filenames and then just made it so that it doesn't load the dataset again. Then I added the option as well to just change the filename instead ("Allow duplicate filenames on").

But I do have to admit that the use-case of this particular setting is very narrow. I'm not completely opposed to removing it, or maybe change the wording a bit. I did fix the bug however, although the fix is a bit ugly code-wise, it does work for now (on a similar note, I will start to add docstrings and comments to the code-base soon as well).

 If I decide to keep it in, I should also add a toast in the "Add equation" window. But again, I'm a bit torn about keeping it in at all. I do see arguments for just removing the functionality all-together.

**Comment by SjoerdB93** [comment]
In fact, the more I think about it, the more I'm leaning towards getting rid of this config setting. I have difficulties finding a clear use-case, it may introduce confusion, and it only adds to clutter (both in the settings and in the code-base).

**Comment by SjoerdB93** [comment]
Pushed a slightly less hacky fix. Now it works as it did previously. Not sure when I broke it.

**Comment by cmkohnen** [comment]
Thanks for the explanation!
> as I use the filename as dictionary key
Maybe a clean solution would be to have a different key e. g. a numerical identifier or UUID

> If I decide to keep it in, I should also add a toast in the "Add equation" window in case it refuses to add an equation due to this setting. 
That was actually, what I was trying to accomplish, only to find out, that the feature wasn't working as intuitively expected.
Will create a pr in a moment

However I think this functionality either needs to be changed or reworded.
I would think enabling this means having two rows of data bearing the same name but with a different color, different values etc.
Disabling would (like in many other apps) append a number in brackets (like if it were currently enabled)
I would suggest rewording to something like "automatically rename items if they already exist"

**Comment by cmkohnen** [comment]
As for a use case: With the changed wording, this could aid in automating workflows, where files are automatically put in a folder on the filesystem, mass imported and maybe some have been imported before, thereby either being skipped or at the users choice added for manual deletion, after choosing which one to keep (maybe the new one has updated data)

**Comment by SjoerdB93** [comment]
> Maybe a clean solution would be to have a different key e. g. a numerical identifier or UUID

A unique identifier would work in a potential future implementation yes. This would also potentially allow people to use exactly the same filename for some reason. I should check how to generate a UUID and how much it messes with the code base, but it sounds relatively easy to implement.

That would have implications for this setting though, and it would probably mean it should be changed to a dropdown option, with what to do with existing file filenames. With the following options:
- Alllow the import of existing filenames
- Skip existing filenames
- Rename existing filenames 

Regardless it would need a rename as I can totally see how one could come to your interpretation. "Rename filenames" as title, with your suggested text as subtitle would already be a big instant improvement.

**Comment by cmkohnen** [comment]
obsolete as of https://github.com/SjoerdB93/Graphs/pull/28
